### DISCUSSION TITLE
Arbitrary Frame Rate/120FPS Support

### DISCUSSION THREAD

**Discussion by XNINEZERO** [open-goal/jak-project#3716] [Ideas]
So I've been playing around a bit with TPL, and am really impressed at the conversion.  Looking at the bug reports and past dozen changelogs here on GitHub, it seems work has shifted to JII and III and TPL seems to be in a good enough state to leave it be.

My inquiry concerns the high frame rate support.  The game is very insistent on its own manual frame rate cap - running an external limiter via Special K/RTSS/NVCP will induce judder, even on a GSYNC panel - as will selecting the maximum available "150" FPS and running it at native refresh rates lower than that (in my case, both 120 and 144 FPS).  Only the game's internal limiter (under Frame Rate - Experimental) functions properly - either when it's set to 100 or the standard 60.  I can only assume that this limiter governs not only the frame rate, but also the actual tick rate that the game and its physics update at.

This is fine, and 100 FPS is obviously a more than satisfactory way to play the game.  I'm just wondering why the interval of 60/100/150 was chosen, and whether this will become a slider in the future.

Small thing, of course, but I figured since OpenGOAL is still in active development I'd ask.

Cheers, and thanks for the great work.

**Comment by Hat-Kid** [comment]
This is an engine limitation with Jak 1 - unlike the later games (which have better support for variable frame rates), it only allows frame rates that are divisors of the tick rate it runs at (300 ticks per second).

**Reply by XNINEZERO** [reply]
I suspected something as much.  Thanks for the info.
